Clarify the handling of `agg_param` to close https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/267 (#282)
ietf-wg-ppm/draft-ietf-ppm-dap: draft-ietf-ppm-dap.md

includes the following parameters:

* The type of each measurement.
* The aggregation function to compute (e.g., sum, mean, etc.) and optional

>   task-constant parameters for the aggregation function.

Reviewer: This parameter changes in Poplar, so I don't think this definition is correct.

Author: That is actually the point I wanted to make here. Previously it looked like the aggregation parameter (the thing that changes in Poplar) was part of the task definition. Since it changes, that's not possible. But of course other things that do not change can be given here.

Reviewer: Yeah, the current language seems wrong too. Maybe we could say something like this?

```
* The aggregation function to compute (e.g., sum, mean, etc.) and a way to
  produce optional parameters for the aggregation function.
```

Author: Maybe we should just decomplicate things and remove it entirely and assume that this kind of thing is part of "the function".

Reviewer 2: I think removing it entirely is the right way to go.

uploaded by the client. The `encrypted_input_share` field is the `HpkeCiphertext`
whose index in `Report.encrypted_input_shares` is equal to the index of the aggregator
in the task's `aggregator_endpoints` to which the AggregateInitializeReq is being sent.

> The `agg_param` field is an opaque, VDAF-specific aggregation parameter which is constant per collection flow.

Reviewer: ```suggestion
The `agg_param` field is an opaque, VDAF-specific aggregation parameter provided during a collection flow.
```

> Note that this is the same value as in `CollectReq` (see {{collect-init}} below) and `AggregateShareReq` (see {{collect-aggregate}} below).

Reviewer: Can we move this note down to these sections? That is, when describing `CollectReq`, we can place a reference back to this section. Likewise for `AggregateShareReq`.

Author: Done.

uploaded by the client. The `encrypted_input_share` field is the `HpkeCiphertext`
whose index in `Report.encrypted_input_shares` is equal to the index of the aggregator
in the task's `aggregator_endpoints` to which the AggregateInitializeReq is being sent.
The `agg_param` field is an opaque, VDAF-specific aggregation parameter which is constant per collection flow.
Note that this is the same value as in `CollectReq` and `AggregateShareReq`.

> Bandwith can be optimized by referencing the value from `AggregateInitializeReq` in `AggregateShareReq`.

Reviewer: While tempting, I don't think this is good advice. The leader doesn't necessarily know that helper will know how to interpret `agg_param` in an `AggregateShareReq` as a reference. In any case, it seems like premature optimization to me, so I'd suggest removing it.

Author: tbh I am not quite sure why we are sending it twice. Maybe elaborating on that here would be more useful.

Reviewer 2: The agg param is needed here because this the first time the Helper learns its value. The AggregateShareReq message isn't sent until later, once the batch is ready to be collected.

If we wanted to save bandwidth, we could consider replacing AggregateShareReq.agg_param with something like SHA-256(agg_param). However I agree that this is premature and can wait until someone has implemented Poplar1.
